### PR TITLE
Winograd RxSf2x3 solver V21_1_2 update

### PR DESCRIPTION
**Pull request by Slimakanzer**
Winograd V21_1_2 changes compared to #175 :
* Fixed bugs
* No performance regression compared to the [previous version](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/175)
* Supports extended cases set (solver's **IsApplicable** will be updated in the next PR)
* Tensor offset support (64 bit byte offsets)

### CODE REVIEW

File: src/kernels/Conv_Winograd_v21_1_2_metadata.inc [hunk 1]
```diff
@@ -71,28 +71,32 @@ amdhsa.kernels:
...
-    - { .size: 4, .offset: 156, .value_kind: by_value, .value_type: i32, .name: f_G_stride }
-    - { .size: 4, .offset: 160, .value_kind: by_value, .value_type: i32, .name: o_G_stride }
-    - { .size: 8, .offset: 168, .value_kind: hidden_global_offset_x, .value_type: i64 }
-    - { .size: 8, .offset: 176, .value_kind: hidden_global_offset_y, .value_type: i64 }
-    - { .size: 8, .offset: 184, .value_kind: hidden_global_offset_z, .value_type: i64 }
-    - { .size: 8, .offset: 192, .value_kind: hidden_none,   .value_type: i8 }
-    - { .size: 8, .offset: 200, .value_kind: hidden_none,   .value_type: i8 }
-    - { .size: 8, .offset: 208, .value_kind: hidden_none,   .value_type: i8 }
+    - { .size: 8, .offset: 104, .value_kind: by_value, .value_type: i64, .name: d_offset }
+    - { .size: 8, .offset: 112, .value_kind: by_value, .value_type: i64, .name: f_offset }
```
Comment (Kirpich30000) [inline src/kernels/Conv_Winograd_v21_1_2_metadata.inc:75]: Please explicitly add reserved2 field. See latest updates of the winograd shader.

File: src/solver/conv_winoRxS_f2x3.cpp [hunk 2]
```diff
@@ -513,8 +512,13 @@ ConvBinWinogradRxSf2x3::GetSolution(const ConvolutionContext& params,
...
         constexpr int L_F_NKC_STRIDES   = 1 << 9;
         constexpr int L_F_GROUP_STRIDES = 1 << 10;
-        int reserved                    = 0;
-        int* reserved_ptr               = nullptr;
+        // constexpr int L_F_FORCE_FILTER_TRAVERSE_MODE  = 1 << 11;
+        // constexpr int L_F_FILTER_TRAVERSE_DUAL  = 1 << 12;
+        // constexpr int L_F_TENSOR_OFFSETS  = 1 << 13;
+        // constexpr int L_F_USE_EXTENDED_FLAGS_64  = 1 << 15;
+        int reserved             = 0;
+        long int reserved_offset = 0;
```
Comment (Kirpich30000) [inline src/solver/conv_winoRxS_f2x3.cpp:520]: better to use uint64_t

File: src/solver/conv_winoRxS_f2x3.cpp [hunk 3]
```diff
@@ -416,11 +416,10 @@ ConvBinWinogradRxSf2x3::GetSolution(const ConvolutionContext& params,
     if(!IsWarned)
     {
         if(params.GetStream().GetMaxComputeUnits() > MAX_CU_LIMIT)
-            MIOPEN_LOG_WE(SolverDbId(*this) << ": GPU has "
-                                            << params.GetStream().GetMaxComputeUnits()
-                                            << "CUs, but this solver supports max "
-                                            << MAX_CU_LIMIT
-                                            << "and thus may show sub-optimal performance.");
+            MIOPEN_LOG_WE(SolverDbId(*this)
```
Comment (Kirpich30000) [inline src/solver/conv_winoRxS_f2x3.cpp:419]: Why was it reformatted?

File: src/solver/conv_winoRxS_f2x3.cpp [hunk 4]
```diff
@@ -267,8 +267,8 @@ bool PerformanceConfigConvBinWinogradRxSf2x3::IsValid(const ConvolutionContext&
     return true;
 }
 
-inline bool PerformanceConfigConvBinWinogradRxSf2x3::
-operator==(const PerformanceConfigConvBinWinogradRxSf2x3& other) const
+inline bool PerformanceConfigConvBinWinogradRxSf2x3::operator==(
+    const PerformanceConfigConvBinWinogradRxSf2x3& other) const
```
Comment (Kirpich30000) [inline src/solver/conv_winoRxS_f2x3.cpp:271]: Why was it reformatted?

**Review by Kirpich30000 — CHANGES_REQUESTED** [review]

**Comment by Kirpich30000** [comment]
Few more features should be added in following PRs. Please create sepparate issues to support:
- gfx10
- filter traverse modes
- extended applicability with multiple dispatches

**Comment by atamazov** [comment]
@Slimakanzer Please use drafts instead of `work_in_progress` label. 
image

File: src/kernels/Conv_Winograd_v21_1_2_metadata.inc [hunk 5]
```diff
@@ -71,28 +71,33 @@ amdhsa.kernels:
     - { .size: 4, .offset:  84, .value_kind: by_value, .value_type: i32, .name: out_w }
     - { .size: 8, .offset:  88, .value_kind: global_buffer, .value_type: f32, .name: bias_addr,    .address_space: global, .is_const: true }
```
Comment (Kirpich30000) [inline src/kernels/Conv_Winograd_v21_1_2_metadata.inc:72]: BTW, why only `bias_addr` is marked as `.is_const: true`
Let's make `in` and `weights` const also

**Review by Kirpich30000 — APPROVED** [review]
LGTM

**Pull request merged**